app/Rules/AccountIdRule.php

```diff
@@ -7,25 +7,57 @@
 
 class AccountIdRule implements Rule
 {
+    /** @var array */
+    private $blacklistedAccountIds;
+
+    /** @var AccountId */
+    private $blacklistedAccountId;
+
+    public function __construct(array $blacklistedAccountIds = [])
+    {
+        $this->blacklistedAccountIds = $blacklistedAccountIds;
+    }
+
     /**
      * Determine if the validation rule passes.
      *
      * @param  string  $attribute
      * @param  mixed  $value
      * @return bool
      */
-    public function passes($attribute, $value)
+    public function passes($attribute, $value): bool
     {
-        return AccountId::isValid($value);
+        if (!AccountId::isValid($value)) {
+            return false;
+        }
+
+        $accountId = new AccountId($value, false);
+
+        foreach ($this->blacklistedAccountIds as $blacklistedAccountId) {
+            if ($accountId->equals($blacklistedAccountId)) {
+                $this->blacklistedAccountId = $blacklistedAccountId;
+
+                return false;
+            }
+        }
+
+        return true;
     }
 
     /**
      * Get the validation error message.
      *
      * @return string
      */
-    public function message()
+    public function message(): string
     {
+        if (null !== $this->blacklistedAccountId) {
+            return sprintf(
+                'The :attribute must be different than blacklisted (%s)',
+                $this->blacklistedAccountId->toString()
```

Comment:
Please use `:input` value instead of assigning an additional variable. 
If from some reasons it will not work try to create a message inside `passes` method and just use it in the `message` method`.